packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1879,6 +1881,11 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       value: labelIsInitiallyFloating ? 1.0 : 0.0,
     );
     _floatingLabelController.addListener(_handleChange);
+    _floatingLabelAnimation = CurvedAnimation(
+      parent: _floatingLabelController,
+      curve: _kTransitionCurve,
+      reverseCurve: _kTransitionCurve.flipped,
```

Comment:
I'm not super familiar with Android, but from https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/textfield/TextInputLayout.java#L4150-L4168, it seems the curve is always `AnimationUtils.FAST_OUT_SLOW_IN_INTERPOLATOR`?

Same question for the change made to `_borderAnimation`.

Reply:
If the flipped curve is not used for the reverse, the curve actually reversed on unfocus, something like slowOutFastIn.
I don't know it's normal behavior or not, but this is what actually happens and can be seen visually in slow motion.